civiform/civiform
# [Program Migration] Support export and import of program definitions

### Description

This PR does a few things to let us export and import program definitions:
 - Updates the `ProgramDefinition` class so that it can be serialized into JSON and deserialized from JSON back into a `ProgramDefinition` object.
 - Updates `AdminExportController` to take the selected program, serialize it into JSON, and download that JSON file.
 - Updates `AdminImportController` to read the uploaded JSON file, deserialize it into a `ProgramDefinition` object, and display the program's blocks and questions on screen.

## Release notes

### Checklist

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)


#### User visible changes


### Instructions for manual testing

1. Enable the PROGRAM_MIGRATION_ENABLED flag.
2. As an admin, open the Export tab.
3. Choose a program and click Download.
4. Open the Import tab.
5. Upload the file you just downloaded and click Upload.
6. Verify the page updates and shows the program you downloaded with its blocks and questions. Note that we only show the question IDs right now. A future PR will show all the question data.

### Issue(s) this completes

Part of epic #7087
